distributed/tests/test_cancelled_state.py

```diff
@@ -294,30 +289,12 @@ def block_execution(lock):
         # to be recomputed on B
         await s.remove_worker(a.address, stimulus_id="foo", close=False, safe=True)
 
-        while not b.state.tasks[fut1.key].state == "resumed":
-            await asyncio.sleep(0.01)
+        await wait_for_state(fut1.key, "resumed", b, interval=0)
```

Comment:
I'm a bit surprised this works without the lock above. At the very least without the lock this might become quite flaky. What prompted you to refactoring this test?

Reply:
I paired with @crusaderky on refactoring this test, it was failing every run with the new code because the `while not b.state.tasks[fut1.key].state == "resumed": await asyncio.sleep(0.01)` missed the key going into the "resumed" state.